src/screens/Settings/stacks/timer/Timer.tsx

```diff
@@ -0,0 +1,115 @@
+
+    const setSleeper = useCallback((value: number, timeType: string) => {
+        if (timeType == 'Hours' && value > 0) {
+            dispatch(setSleepTime(value * 3600));
+        } else if (timeType == 'Minutes' && value > 0) {
+            dispatch(setSleepTime(value * 60));
+        }
+    }, [navigation, dispatch]);
+
+    const getTime = () => {
+        let hours = 0;
+        let minutes = 0;
+        if (!Number.isNaN(sleepTime)) {
+            hours = Math.round(sleepTime / 3600);
+            const timeRemaining = sleepTime % 3600;
+
+            if (timeRemaining >= 60) {
+                minutes = Math.round(timeRemaining / 60);
+            }
+        }
+
+        return `${hours} hrs ${minutes} min`;
+    }
+
+    const handleEnabledSleeper = useCallback((value: boolean) => {
+        dispatch(setEnabledSleeper(value));
+        setEnableSleeper(value);
+
+        // If value is true sleeper has been enabled, pause then play tack
+        // to trigger play state and start sleeper timer
+        if (value) {
+            TrackPlayer.pause()
+            TrackPlayer.play()
+        }
+    }, [dispatch]);
+
+    return (
+        <Container>
+            <InputContainer>
+                <Text>Set Sleep Timer. Time Set Previously: {getTime()}</Text>
+                <View style={timerStyles.checkbox}>
+                    <CheckBox
+                        value={enableSleeper}
+                        onValueChange={(value) => handleEnabledSleeper(value)}
+                    />
+                    <Text> Enable Sleeper</Text>
+                </View>
+                <View style={enabledSleeper ? timerStyles.timerSetting : timerStyles.timerSettingsDisabled}>
+                    <View style={timerStyles.timer}>
+                        <View style={timerStyles.timeInput}>
+                            <Text>H:</Text>
+                            <Input
+                                placeholder='60'
+                                editable={enabledSleeper}
+                                style={timerStyles.input}
+                                keyboardType='phone-pad'
+                                onChangeText={(value: string) => {
+                                    setSleeper(parseInt(value), 'Hours');
+                                }}
+                            />
+                        </View>
+                        <View style={timerStyles.timeInput}>
+                            <Text>M:</Text>
+                            <Input
+                                placeholder='60'
+                                editable={enabledSleeper}
+                                style={timerStyles.input}
+                                keyboardType='phone-pad'
+                                onChangeText={(value: string) => {
+                                    setSleeper(parseInt(value), 'Minutes');
+                                }}
+                            />
+                        </View>
```

Comment:
I think there are better input methods for setting a duration. Text boxes are small and aren't always clear on the current state. I recommend looking at [react-native-datetimepicker](https://github.com/react-native-datetimepicker/datetimepicker) for selecting the duration. As far as I can see, iOS has a `countdown` display option.

Reply:
In my opinion a timer would be more efficient, that way a user will not constrained to a specific time. When enabled it will take an interval of the set time to pause the music. This can be enabled at any time of the day.

With date and time set the music or audio will pause even if they have not slept at that time they set.

Reply:
I have used this library: [react-native-modal-datetime-picker](https://github.com/mmazzarolo/react-native-modal-datetime-picker) with what you had suggested: [datetimepicker](https://github.com/react-native-datetimepicker/datetimepicker) to implement the date picker. The community react native version had a bug I could not find a workaround for.